### DISCUSSION TITLE
Is thare any way to "freeze" .der files in the firmware

### DISCUSSION THREAD

**Discussion by Hristo-Nikodimov-Nenkov** [micropython/micropython#14166] [Using MicroPython]
Greetings.

I'm using Microdot as backend server on Pico W and I was wandering: Is there any way to "bake" the .der ( cert.der & key.der ) into the firmware.

Any and all help is more than welcome.

**Comment by andrewleech** [comment]
There's no "official" way to do this yet until https://github.com/micropython/micropython/pull/8381 is finished.

Until then there's been a number of discussions about other options likehttps://github.com/orgs/micropython/discussions/10707 or https://github.com/micropython/micropython/issues/5394

**Reply by Josverl** [reply]
And also a nice summary in :

https://github.com/micropython/micropython/discussions/14144#discussioncomment-8866194


(It helps to search before asking, it saves time and sometimes you learn stuff you can use)

**Reply by andrewleech** [reply]
For reference I compiled some of these notes here: https://github.com/micropython/micropython/wiki/Bundling-and-deploying-application-code-and-files which is linked from the wiki tips and tricks page
